tests/test_data/test_summary.py

```diff
@@ -3,56 +3,75 @@
-            num_of_entries = sum(1 for line in tsv)
-            observed_name_to_count[unique_tsv] = num_of_entries
-
-    with open(_SUMMARY, "r", encoding="utf-8") as lang_summary:
-        summary_files = [line.rstrip().split("\t")[0] for line in lang_summary]
-
-    for summary_file in summary_files:
-        assert (
-            summary_file in observed_name_to_count
-        ), f"{summary_file} in data/tsv_summary.tsv but not in data/tsv"
+_PHONES_SUMMARY = os.path.join(_REPO_DIR, "data/phones/phones_summary.tsv")
+_PHONES_DIRECTORY = os.path.join(_REPO_DIR, "data/phones")
 
 
 def test_language_data_matches_summary():
     """Check if each TSV in data/tsv is present in data/tsv_summary.tsv
-    and if the number of entries in each TSV matches its listed number
-    of entries in data/tsv_summary.tsv.
+    (and vice-versa) as well as if the number of entries in each TSV in
+    data/tsv matches its listed number of entries in data/tsv_summary.tsv.
 
     (Basically checks whether generate_tsv_summary.py has been run.)
     """
-    name_count_dict = {}
-    with open(_SUMMARY, "r", encoding="utf-8") as lang_summary:
-        vals = [line.rstrip().split("\t") for line in lang_summary]
-        for val in vals:
-            name_count_dict[val[0]] = int(val[-1])
+    name_to_count = {}
+    with open(_TSV_SUMMARY, "r", encoding="utf-8") as lang_summary:
+        languages_data = [line.rstrip().split("\t") for line in lang_summary]
+        for language in languages_data:
+            name_to_count[language[0]] = int(language[-1])
 
     for unique_tsv in os.listdir(_TSV_DIRECTORY):
         with open(
             f"{_TSV_DIRECTORY}/{unique_tsv}", "r", encoding="utf-8"
         ) as tsv:
             num_of_entries = sum(1 for line in tsv)
-        assert unique_tsv in name_count_dict, (
+        assert unique_tsv in name_to_count, (
             f"{unique_tsv} in data/tsv but not in " "data/tsv_summary.tsv"
         )
-        assert name_count_dict[unique_tsv] == num_of_entries, (
+        assert name_to_count[unique_tsv] == num_of_entries, (
             f"Number of entries in {unique_tsv} does not match "
             "number of entries in data/tsv_summary.tsv."
         )
+        del name_to_count[unique_tsv]
+    assert len(name_to_count) == 0, (
+        "The following TSVs are listed in data/tsv_summary.tsv "
+        "but could not be found in data/tsv: "
+        f"{[name for name in name_to_count.keys()]}"
+    )
+
+
+def test_phones_data_matches_summary():
+    """Check if each .phones file in data/phones is present in
+    data/phones/phones_summary.tsv and if the number of phones in each .phones
+    file matches its listed number of phones in data/phones_summary.tsv.
+
+    (Basically checks whether generate_phones_summary.py has been run.)
+    """
+    name_to_count = {}
+    with open(_PHONES_SUMMARY, "r", encoding="utf-8") as phones_summary:
+        phones_data = [line.rstrip().split("\t") for line in phones_summary]
```

Comment:
This is pure style, and not a particularly prescribed sort of style, but I think this would have been better as:

```
for line in phones_summary:
   languages = line.rstrip().split(\"t)
   for language in languages: ...
```

or even get rid of the intermediate `languages` varable if you want.

Reply:
I went a bit crazy with the list comprehensions here, so I've dialed it back.